api-doc/openapi.yaml

```diff
@@ -1,48 +1,514 @@
 openapi: 3.0.3
 info:
-  version: 0.0.1
+  version: 0.0.3.
   title: Vauhtijuoksu API
 paths:
-  /target:
+  /donations:
     get:
       responses:
-        200:
-          $ref: '#/components/responses/TargetOk'
+        '200':
+          $ref: '#/components/responses/DonationsAllOk'
+  /donation/{donationsId}:
+    parameters:
+      - name: donationsId
+        in: path
+        required: true
+        description: Donation id (uuid).
+        schema:
+          type: integer
+          format: int64
+          minimum: 1
+    get:
+      responses:
+        '200':
+          $ref: '#/components/responses/DonationsOk'
+    post:
+      security:
+        - ApiKeyAuth: []
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Donations'
+      responses:
+        '200':
+          $ref: '#/components/responses/OkEmpty'
+    delete:
+      security:
+        - ApiKeyAuth: []
+      responses:
+        '200':
+          $ref: '#/components/responses/OkEmpty'
+  /gamedata:
+    get:
+      responses:
+        '200':
+          $ref: '#/components/responses/GameDataAllOk'
+  /gamedata/{gamedataId}:
+    parameters:
+      - name: gamedataId
+        in: path
+        required: true
+        description: Game data id (uuid).
+        schema:
+          type: integer
+          format: int64
+          minimum: 1
+    get:
+      responses:
+        '200':
+          $ref: '#/components/responses/GameDataOk'
+    post:
```

Comment:
Are these for modifying or creating data? For former, I'd suggest maybe just PATCH, for latter I think it should be just `/gamedata/`

Reply:
I thought that this would upsert data, so creating and modifying. I did not catch what did you mean by  'for latter I think it should be just /gamedata/', but separate PATCH can be added if it makes more sense.

Reply:
IIRC, creating a new resource should be either POST to `/gamedata/` or PUT to `/gamedata/id`. I prefer former, so client doesn't need to generate an id.

Furthermore, updating data with POST should always have complete payload, so for updating data I prefer PATCH.

Reply:
This is now implemented. I added also PUT for endpoints that would benefit client generated ids.